pandas/tests/indexes/multi/test_setops.py

```diff
@@ -337,7 +339,9 @@ def test_union_sort_other_incomparable():
     idx = pd.MultiIndex.from_product([[1, pd.Timestamp("2000")], ["a", "b"]])
 
     # default, sort=None
-    result = idx.union(idx[:1])
+    warn = None if _is_numpy_dev else RuntimeWarning
+    with tm.assert_produces_warning(warn):
+        result = idx.union(idx[:1])
```

Comment:
> Can you test this from an end user perspective?

@WillAyd Isn't this already testing from an end user perspective?

Reply:
Ah sorry overlooked this one